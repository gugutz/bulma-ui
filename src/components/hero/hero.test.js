import React from 'react'
import renderer from 'react-test-renderer'
import Hero from './hero'

describe('<Hero /> component', () => {
  it('should render correctly', () => {
    const tree = renderer
      .create(
        <Hero>
          <Hero.Head>header</Hero.Head>
          <Hero.Body>body</Hero.Body>
          <Hero.Foot>footer</Hero.Foot>
        </Hero>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should support isBold prop', () => {
    const tree = renderer
      .create(
        <Hero isBold>
          <Hero.Body>text</Hero.Body>
        </Hero>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  const colors = [
    'primary',
    'info',
    'success',
    'warning',
    'danger',
    'light',
    'dark'
  ]

  for (const color of colors) {
    it('should support color prop with ' + color, () => {
      const tree = renderer
        .create(
          <Hero color={color}>
            <Hero.Body>text</Hero.Body>
          </Hero>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  }

  const sizes = ['medium', 'large', 'fullheight']

  for (const size of sizes) {
    it('should support size prop with ' + size, () => {
      const tree = renderer
        .create(
          <Hero size={size}>
            <Hero.Body>text</Hero.Body>
          </Hero>
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  }
})
