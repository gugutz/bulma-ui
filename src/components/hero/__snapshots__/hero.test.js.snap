// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hero /> component should render correctly 1`] = `
<div
  className="hero"
>
  <div
    className="hero-head"
  >
    header
  </div>
  <div
    className="hero-body"
  >
    body
  </div>
  <div
    className="hero-foot"
  >
    footer
  </div>
</div>
`;

exports[`<Hero /> component should support color prop with danger 1`] = `
<div
  className="hero is-danger"
>
  <div
    className="hero-body"
  >
    text
  </div>
</div>
`;

exports[`<Hero /> component should support color prop with dark 1`] = `
<div
  className="hero is-dark"
>
  <div
    className="hero-body"
  >
    text
  </div>
</div>
`;

exports[`<Hero /> component should support color prop with info 1`] = `
<div
  className="hero is-info"
>
  <div
    className="hero-body"
  >
    text
  </div>
</div>
`;

exports[`<Hero /> component should support color prop with light 1`] = `
<div
  className="hero is-light"
>
  <div
    className="hero-body"
  >
    text
  </div>
</div>
`;

exports[`<Hero /> component should support color prop with primary 1`] = `
<div
  className="hero is-primary"
>
  <div
    className="hero-body"
  >
    text
  </div>
</div>
`;

exports[`<Hero /> component should support color prop with success 1`] = `
<div
  className="hero is-success"
>
  <div
    className="hero-body"
  >
    text
  </div>
</div>
`;

exports[`<Hero /> component should support color prop with warning 1`] = `
<div
  className="hero is-warning"
>
  <div
    className="hero-body"
  >
    text
  </div>
</div>
`;

exports[`<Hero /> component should support isBold prop 1`] = `
<div
  className="hero is-bold"
>
  <div
    className="hero-body"
  >
    text
  </div>
</div>
`;

exports[`<Hero /> component should support size prop with fullheight 1`] = `
<div
  className="hero is-fullheight"
>
  <div
    className="hero-body"
  >
    text
  </div>
</div>
`;

exports[`<Hero /> component should support size prop with large 1`] = `
<div
  className="hero is-large"
>
  <div
    className="hero-body"
  >
    text
  </div>
</div>
`;

exports[`<Hero /> component should support size prop with medium 1`] = `
<div
  className="hero is-medium"
>
  <div
    className="hero-body"
  >
    text
  </div>
</div>
`;
